common/amount.c

```diff
@@ -49,6 +52,9 @@ const char *fmt_amount_sat_btc(const tal_t *ctx,
 			       const struct amount_sat *sat,
 			       bool append_unit)
 {
+	if (sat->satoshis == 0)
+		return tal_fmt(ctx, append_unit ? "0btc" : "0");
```

Comment:
since these aren't runtime variable length strings, you don't need `tal_fmt` here

```
return append_unit ? "0btc" : "0";
```